src/main.rs

```diff
@@ -36,15 +55,23 @@ struct MyApp {
 }
 
 impl MyApp {
-    fn new() -> MyApp {
+    fn new(path: Option<PathBuf>) -> MyApp {
         let (sender, receiver) = mpsc::channel();
 
-        MyApp {
+        let app = MyApp {
             app_data: None,
             control_data: ControlData::default(),
             sender,
             receiver,
+        };
+
+        if let Some(path) = path {
+            app.sender
+                .send(Message::LoadDiff(path))
+                .expect("Channel closed unexpectedly!");
         }
```

Comment:
This can be executed before the App struct is created. That way no temporary variable is needed and the struct can be returned directly.

Reply:
Thanks I didn't se that `sender` was created before `app`